File: VB/CustomEditor/MyCheckedComboBoxEditViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.Registrator
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsApplication1
	Public Class MyCheckedComboBoxEditViewInfo
		Inherits ButtonEditViewInfo
		Implements IHeightAdaptable
		Public Sub New(ByVal item As RepositoryItem)
			MyBase.New(item)

		End Sub

		Protected Overrides Overloads Sub CalcTextSize(ByVal g As Graphics, ByVal useDisplayText As Boolean)
			MyBase.CalcTextSize(g, True)
		End Sub

		#Region "IHeightAdaptable Members"

		Private Function CalcHeight(ByVal cache As DevExpress.Utils.Drawing.GraphicsCache, ByVal width As Integer) As Integer Implements IHeightAdaptable.CalcHeight
			CalcTextSize(cache.Graphics)
			Return TextSize.Height
		End Function

		#End Region
	End Class
End Namespace

File: CS/CustomEditor/MyCheckedComboBoxEditViewInfo.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.Registrator;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsApplication1
{
    public class MyCheckedComboBoxEditViewInfo : ButtonEditViewInfo, IHeightAdaptable
    {
        public MyCheckedComboBoxEditViewInfo(RepositoryItem item)
            : base(item)
        {

        }

        protected override void CalcTextSize(Graphics g, bool useDisplayText)
        {
            base.CalcTextSize(g, true);
        }

        #region IHeightAdaptable Members

        int IHeightAdaptable.CalcHeight(DevExpress.Utils.Drawing.GraphicsCache cache, int width)
        {
            CalcTextSize(cache.Graphics);
            return TextSize.Height;
        }

        #endregion
    }
}
